### DISCUSSION TITLE
<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies

### DISCUSSION THREAD

**Discussion by conradh** [vercel/next.js#35877] [Help] [answered]
Hi i have exactly same error as closed Issue #27650 so need help

I have been trying to get 1 project work for 2 days now. I have wiped all traces of nodejs, reinstalled it, cloned example nextjs projects from scratch, and this error occurs on "npm run build"

> Some kind of error happened... You need to enable debug logging to see the error:
> 
> ```js
> // next.config.js
> module.exports = {
>   webpack(config) {
>     config.infrastructureLogging = { debug: /PackFileCache/ }
>     return config;
>   }
> }
> ```

Here's what I got on my console after enabling it:

PS E:\React Projects\testapp> yarn dev
yarn run v1.22.11
$ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
    [webpack.cache.PackFileCacheStrategy] No pack exists at E:\React Projects\testapp\.next\cache\webpack\client-development.pack: Error: ENOENT: no such f
ile or directory, stat 'E:\React Projects\testapp\.next\cache\webpack\client-development\index.pack'
<t> [webpack.cache.PackFileCacheStrategy] restore cache container: 47.1833 ms
    [webpack.cache.PackFileCacheStrategy] No pack exists at E:\React Projects\testapp\.next\cache\webpack\server-development.pack: Error: ENOENT: no such f
ile or directory, stat 'E:\React Projects\testapp\.next\cache\webpack\server-development\index.pack'
<t> [webpack.cache.PackFileCacheStrategy] restore cache container: 44.0038 ms
    [webpack.cache.PackFileCacheStrategy] Pack got invalid because of write to: ResolverCachePlugin|normal|dependencyType=|esm|path=|E:\React Projects\test
app|request=|E:\React Projects\testapp\node_modules\@next\react-refresh-utils\runtime.js
    [webpack.cache.PackFileCacheStrategy] Storing pack...
    [webpack.cache.PackFileCacheStrategy] Capturing build dependencies... (E:\React Projects\testapp\next.config.js, E:\React Projects\testapp\node_modules
\next\dist\compiled\, E:\React Projects\testapp\node_modules\next\dist\build\webpack\loaders\next-client-pages-loader.js, E:\React Projects\testapp\node_mo
dules\@next\react-refresh-utils\loader.js, E:\React Projects\testapp\node_modules\next\dist\build\babel\loader\index.js, E:\React Projects\testapp\node_mod
ules\next\dist\build\webpack\loaders\next-style-loader\index.js, E:\React Projects\testapp\node_modules\next\dist\compiled\css-loader\cjs.js, E:\React Proj
ects\testapp\node_modules\next\dist\compiled\postcss-loader\cjs.js)
event - compiled successfully
    [webpack.cache.PackFileCacheStrategy] Pack got invalid because of write to: ResolverCachePlugin|normal|dependencyType=|esm|path=|E:\React Projects\test
app|request=|private-next-pages/_app.js
    [webpack.cache.PackFileCacheStrategy] Storing pack...
    [webpack.cache.PackFileCacheStrategy] Capturing build dependencies... (E:\React Projects\testapp\next.config.js, E:\React Projects\testapp\node_modules
\next\dist\compiled\, E:\React Projects\testapp\node_modules\next\dist\build\babel\loader\index.js, E:\React Projects\testapp\node_modules\next\dist\compil
ed\ignore-loader\index.js)
<t> [webpack.cache.PackFileCacheStrategy] resolve build dependencies: 2174.4305 ms
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects\testapp: Error: EI
NVAL: invalid argument, read
<t> [webpack.cache.PackFileCacheStrategy] snapshot build dependencies: 10.1572 ms
<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies
    [webpack.cache.PackFileCacheStrategy] Error: Unable to snapshot resolve dependencies
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:77909:13
        at jobError (E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:56110:5)
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:56161:10
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:56957:13
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:130615:5
        at Hook.eval [as callAsync] (eval at create (E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:31934:10), <anonymous>:6:
1)
        at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:31736:14)
        at AsyncQueue._handleResult (E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:130585:21)
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:130568:11
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:56906:12
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects: Error: EINVAL: in
valid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\: Error: EINVAL: invalid argument
, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects\testapp\node_modul
es\@next: Error: EINVAL: invalid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects\testapp\node_modul
es: Error: EINVAL: invalid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects\testapp\node_modul
es\@babel: Error: EINVAL: invalid argument, read
<t> [webpack.cache.PackFileCacheStrategy] resolve build dependencies: 1628.2388 ms
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects: Error: EINVAL: in
valid argument, read
<t> [webpack.cache.PackFileCacheStrategy] snapshot build dependencies: 8.8026 ms
<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies
    [webpack.cache.PackFileCacheStrategy] Error: Unable to snapshot resolve dependencies
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:77909:13
        at jobError (E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:56110:5)
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:56161:10
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:56957:13
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:130615:5
        at Hook.eval [as callAsync] (eval at create (E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:31934:10), <anonymous>:6:
1)
        at AsyncQueue._handleResult (E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:130585:21)
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:130568:11
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:56906:12
        at E:\React Projects\testapp\node_modules\next\dist\compiled\webpack\bundle5.js:29533:16
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects\testapp: Error: EI
NVAL: invalid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\: Error: EINVAL: invalid argument
, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects\testapp\node_modul
es: Error: EINVAL: invalid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of E:\React Projects\testapp\node_modul
es\@babel: Error: EINVAL: invalid argument, read

_Originally posted by @Fangston0 in https://github.com/vercel/next.js/issues/27650#issuecomment-891831607_

**Comment by conradh** [comment]
A simple way to reproduce for me:

1 install latest node.js from scratch:
https://nodejs.org/en/

2. follow instructions on initialize an example project:
https://nextjs.org/learn/basics/create-nextjs-app/setup

3. "npm run build"

```
PS K:\repo\nexttest\nextjs-blog> npm run build

> build
> next build

info  - Checking validity of types
warn  - No ESLint configuration detected. Run next lint to begin setup
info  - Creating an optimized production build ...<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies
info  - Creating an optimized production build
Failed to compile.

Error: EISDIR: illegal operation on a directory, readlink 'K:\repo\nexttest\nextjs-blog\pages\index.js'


> Build failed because of webpack errors
```

**Comment by conradh** [comment]
And this is the extended log from when building my other project (same error as example project in previous post):

K:\repo\project\webapp>npm run build

> react@0.1.0 build
> next build

info  - Checking validity of types

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
info  - Creating an optimized production build ..    [webpack.cache.PackFileCacheStrategy] No pack exists at K:\repo\project\webapp\.next\cache\webpack\client-production.pack: Error: ENOENT: no such file or directory, stat 'K:\repo\project\webapp\.next\cache\webpack\client-production\index.pack'
<t> [webpack.cache.PackFileCacheStrategy] restore cache container: 31.8581 ms
info  - Creating an optimized production build ..    [webpack.cache.PackFileCacheStrategy] Pack got invalid because of write to: Compilation/modules|K:\repo\project\webapp\node_modules\next\dist\build\webpack\loaders\next-client-pages-loader.js?page=%2F_app&absolutePagePath=private-next-pages%2F_app.tsx!
    [webpack.cache.PackFileCacheStrategy] Storing pack...
    [webpack.cache.PackFileCacheStrategy] Capturing build dependencies... (K:\repo\project\webapp\next.config.js, K:\repo\project\webapp\node_modules\next\dist\compiled\, K:\repo\project\webapp\node_modules\next\dist\build\webpack\loaders\next-client-pages-loader.js, K:\repo\project\webapp\node_modules\next\dist\build\webpack\loaders\next-swc-loader.js, K:\repo\project\webapp\node_modules\next\dist\build\webpack\loaders\css-loader\src\index.js, K:\repo\project\webapp\node_modules\next\dist\build\webpack\loaders\postcss-loader\src\index.js, K:\repo\project\webapp\node_modules\next\dist\compiled\mini-css-extract-plugin\loader.js)
<t> [webpack.cache.PackFileCacheStrategy] resolve build dependencies: 48.806 ms
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of K:\repo\project: Error: EINVAL: invalid argument, read
<t> [webpack.cache.PackFileCacheStrategy] snapshot build dependencies: 1.9132 ms
<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies
    [webpack.cache.PackFileCacheStrategy] Error: Unable to snapshot resolve dependencies
        at K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:62463:13
        at jobError (K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:38954:5)
        at K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:39017:10
        at K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:39914:13
        at K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:121794:5
        at Hook.eval [as callAsync] (eval at create (K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:140346:10), <anonymous>:6:1)
        at AsyncQueue._handleResult (K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:121764:21)
        at K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:121747:11
        at K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:39863:12
        at K:\repo\project\webapp\node_modules\next\dist\compiled\webpack\bundle5.js:14502:16
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of K:\repo\project\webapp: Error: EINVAL: invalid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of K:\repo: Error: EINVAL: invalid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of K:\: Error: EINVAL: invalid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of K:\repo\project\webapp\node_modules\@next: Error: EINVAL: invalid argument, read
    [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Error snapshotting file timestamp hash combination of K:\repo\project\webapp\node_modules: Error: EINVAL: invalid argument, read
<t> [webpack.cache.PackFileCacheStrategy] restore cache container: 19.4137 ms
<t> [webpack.cache.PackFileCacheStrategy] check build dependencies: 0.5415 ms
<t> [webpack.cache.PackFileCacheStrategy] restore cache content metadata: 1.0614 ms
    [webpack.cache.PackFileCacheStrategy] starting to restore cache content 0 (90 KiB) because of request to: Compilation/modules|K:\repo\project\webapp\node_modules\next\dist\build\webpack\loaders\next-swc-loader.js??ruleSet[1].rules[2].oneOf[2].use!K:\repo\project\webapp\pages\_app.tsx
<t> [webpack.cache.PackFileCacheStrategy] restore cache content 0 (90 KiB): 6.3061 ms
info  - Creating an optimized production build
Failed to compile.

Error: EISDIR: illegal operation on a directory, readlink 'K:\repo\project\webapp\pages\_app.tsx'


> Build failed because of webpack errors

**Comment by conradh** [comment, marked as answer]
Omg, 2.5 days of problems, and realized that the similar reported issue in #27650 was refering to E:\ and I am having problem on K:\

Solution: Move the project to C:\ and it works!

I leave it up to someone else to find out what the actual error is.

**Reply by cjmcintyre** [reply]
I recently started to get this issue after updating my yarn to the newest 1.22.19 (not sure if that had any cause just the only thing I remember occurring beforehand).. I was on E:\ and moved to C:\ surprisingly this fixed my issue as well.

**Reply by jcourson8** [reply]
If this doesn't work for someone (me), this one did: 
https://github.com/vercel/next.js/discussions/35877#discussioncomment-10963109

**Comment by Stanley-Moukhametzianov** [comment]
I had the same issue but I was on a Mac. I was using node version 18. What worked for me was : 
- npm install -g n
- sudo n stable
This would give you a stable version of node. Then I deleted the node_modules and package-lock.json and ran npm install again. This solved the error for me.

**Reply by webbahadurproject** [reply]
If you are in a Mac and you installed node with brew, make sur you run: 
`brew remove node --force`
before you delete de node_modules and package-lock.json, or otherwise it's not going to use the node version you just installed as probably the one installed by brew is older.

**Comment by imangi** [comment]
is there any solution for this for windows users? I'm having the same issue.

**Reply by imangi** [reply]
I changed my file system to NTFS in windows and then it worked perfectly. Thanks

**Comment by DanielWahl** [comment]
Hey everyone,
I have the same issue on a Debian11 server. I also opened a ticket for it: #47394 
I would appreciate every help.

**Comment by LucasZapico** [comment]
Just figured I callout another case where this error can occur. I ran into this error when I added a new env var to my app in `.env` and my `env.config` but forgot to add it to `next.config.mjs`.

```ts:file=next.config.mjs
/** @type {import('next').NextConfig} */
const nextConfig = withPWA({
  swcMinify: true,
  reactStrictMode: true,
  eslint: {
    dirs: ["src"],
  },
  env: {
    NEXT_PUBLIC_BASE_URL: process.env.BASE_URL || "localhost",
    NEXT_PUBLIC_PORT: process.env.PORT || "3000",
    /**👇 I forgot to add this here**/
    GOOGLE_MAPS_API_KEY: process.env.GOOGLE_MAPS_API_KEY || "",
  },
  feature: {
    webpack5: true,
  },

  webpack: (config) => {
    
    config.resolve.fallback = { ...config.resolve.fallback, fs: false };

    return config;
  },
});
```

**Comment by Mitrajit** [comment]
I got mine solved by changing the node version from 23 to 22.

**Reply by mamlzy** [reply]
You are correct, node 23.0.0 causing this issue. Downgrading back to 22.8.0. Thank you!

**Reply by oxisto** [reply]
Finally... this was the only thing helping me as well.

**Reply by alex-zwingli** [reply]
This instantly fixed my issue too! Wish I would have seen this first instead of researching the above.....

**Reply by ellyfbernardo** [reply]
THANKS!! I was stuck for 3 days.

**Reply by jcourson8** [reply]
This is the way...

**Reply by dpasirst** [reply]
Same problem, this solution worked! I'm guessing there is a better solution that will work with newer versions of node...?
